### PR TITLE
UPSTREAM:<carry> Tolerate worker specific labels and small memory difference while balancing nodegroup

### PR DESCRIPTION
**Pull request by hardikdr**
**What this PR does / why we need it**: CA now tolerates a 128Ki of memory difference and worker-specific labels while balancing the nodegroups. 

**Which issue(s) this PR fixes**:
Fixes [#234](https://github.com/gardener/machine-controller-manager/issues/234)

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Ignore worker specific labels and 128Ki of memory difference while balancing nodegroups.
```

### CODE REVIEW

**Comment by hardikdr** [comment]
With this PR, we are opting for minor customizations in the CA core. We have observed that rebasing the CA to the latest version every time with upstream fixes is a really hard task. 
 - Eg., we need to consume the patch for CA-balancing but seems vendoring has diverged heavily and we can't easily rebase now.
  - We have also seen a need for minor changes in CA to adapt to our needs.  

We anyways need a fresh rebase with cleaner history with upstream as mentioned [here](https://github.com/gardener/autoscaler/issues/24). 
With this fresh rebase and also future iterations we shall take an decision for the PRs/commits prefixed by `<carry>`, whether to replay or revert them.
- It goes without saying that we stay cautious while updating the core, and not make unnecessary changes.  
Hope that's fine.

File: cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go [hunk 1]
```diff
@@ -101,6 +118,12 @@ func IsNodeInfoSimilar(n1, n2 *schedulercache.NodeInfo) bool {
 			if label == kubeletapis.LabelZoneRegion {
 				continue
 			}
+			if label == kubeletapis.LabelGardenerWorkerGroup {
```
Comment (amshuman-kr) [inline cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go]: This is gardener specific change. Whereas the [above](https://github.com/gardener/autoscaler/pull/23/files#diff-0da093ed8e25f6d92d7d334106d395baR79) is an upstream change. Isn't better to keep them in separate commits?

Comment (hardikdr) [reply cluster-autoscaler/utils/nodegroupset/compare_nodegroups.go]: thanks for the review. done.
